analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -60,9 +60,20 @@ public void Execute()
         private IEnumerable<ExplodedNode> ProcessBranching(ExplodedNode node)
         {
             // ToDo: This is a temporary simplification until we support proper branching. This only continues to the next ordinal block
-            if (node.Block.Kind != BasicBlockKind.Exit)
+            if (node.Block.Kind == BasicBlockKind.Exit)
             {
-                yield return new ExplodedNode(cfg.Blocks[node.Block.Ordinal + 1], node.State);
+                NotifyExitReached(node.State);
+            }
+            else
+            {
+                if (node.Block.ContainsThrow())
```

Comment:
Can be simple (at least for now)
```suggestion
            else if (node.Block.ContainsThrow())
```